### PR TITLE
[2273] Keep the parent experiment's filtering and integration settings when subsetting

### PR DESCRIPTION
**Pull request by cosa65**
# Description
 
 
 # Details
 #### URL to issue
 https://biomage.atlassian.net/browse/BIOMAGE-2273
 
 
 #### Link to staging deployment URL (or set N/A)
 N/A (in api)
 
 
 #### Links to any PRs or resources related to this PR
 https://github.com/biomage-org/api/pull/138
 
 
 #### Integration test branch
 master
 
 
 # Merge checklist
 Your changes will be ready for merging after all of the steps below have been completed.
 
 
 ### Code updates
 Have best practices and ongoing refactors being observed in this PR
 - [ ] Migrated any selector / reducer used to the new format.
 
 ### Manual/unit testing
 - [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
 - [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
 - [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.
 
 
 
 ### Integration testing
 **You must check the box below** to run integration tests on the latest commit on your PR branch.
 Integration tests have to pass before the PR can be merged. Without checking the box, your PR
 **will not pass** the required status checks for merging.
 
 - [ ] Started end-to-end tests on the latest commit.
 
 ### Documentation updates
 - [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
 - [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.
 
 ### Optional
 - [ ] Staging environment is unstaged before merging.
 - [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: pipeline-runner/R/gem2s-6-construct_qc_config.R [hunk 1]
```diff
@@ -9,22 +9,17 @@
...
-           unfiltered_samples) {
+construct_qc_config <- function(scdata_list, unfiltered_samples) {
     samples <- names(scdata_list)
 
     config_classifier <-
       add_custom_config_per_sample(
         customize_classifier_config,
         processing_config_template[["classifier"]],
         scdata_list,
-        disable_qc_filters,
```
Comment (aerlaut) [inline pipeline-runner/R/gem2s-6-construct_qc_config.R:27]: why is this removed? Don't we disable this filter for most dataset?

Comment (cosa65) [reply pipeline-runner/R/gem2s-6-construct_qc_config.R:27]: Yes, but now, when we do a subset pipeline, the config is taken from the parent and prepared in that first subset-specific step.

This means that gem2s-6 doesn't need to prepare the config anymore, so this `disable_qc_filters` (which was basically the same as "is a subset pipeline") is no longer necessary because this will never happen.

Either it is a subset and we just pick up the config received from the previous step or it isn't and we generate a normal default config

Comment (aerlaut) [reply pipeline-runner/R/gem2s-6-construct_qc_config.R:27]: Oh ok. I thought this was set to disable the filters in the step (e.g. to disable classifier)

**Review by aerlaut — APPROVED** [review]

File: pipeline-runner/.lintr [hunk 2]
```diff
@@ -1,4 +1,4 @@
 linters: with_defaults(
-   line_length_linter = line_length_linter(80),
+   line_length_linter = line_length_linter(100),
```
[the remark this answers is not in the document]
Comment (gerbeldo) [reply pipeline-runner/.lintr:2]: fine by me. I think that is the `lintr` package default. but R usually has long lines.

File: pipeline-runner/R/subset-1-subset_seurat.R [hunk 3]
```diff
@@ -1,5 +1,36 @@
 library(uuid)
 
+translate_processing_config <- function(parent_processing_config, sample_id_map) {
```
Comment (gerbeldo) [inline pipeline-runner/R/subset-1-subset_seurat.R:3]: this could have a docstring, but it's also pretty self-explanatory

**Review by gerbeldo — APPROVED** [review]
looks good, nothing to add really.

File: pipeline-runner/R/subset-1-subset_seurat.R (hunk 3 above)
Comment (cosa65) [reply pipeline-runner/R/subset-1-subset_seurat.R:3]: Added docstring and changed the name to `subset_processing_config` because it seems clearer
